pkg/slurm/GetLogs.go

```diff
@@ -17,6 +23,121 @@ import (
+func (h *SidecarHandler) GetLogsFollowMode(spanCtx context.Context, podUid string, w http.ResponseWriter, r *http.Request, path string, req commonIL.LogStruct, containerOutputPath string, containerOutput []byte, sessionContext string) error {
+	// Follow until this file exist, that indicates the end of container, thus the end of following.
+	containerStatusPath := path + "/" + req.ContainerName + ".status"
+	// Get the offset of what we read.
+	containerOutputLastOffset := len(containerOutput)
+	sessionContextMessage := GetSessionContextMessage(sessionContext)
+	log.G(h.Ctx).Debug(sessionContextMessage, "Check container status", containerStatusPath, " with current length/offset: ", containerOutputLastOffset)
+
+	var containerOutputFd *os.File
+	var err error
+	for {
+		containerOutputFd, err = os.Open(containerOutputPath)
+		if err != nil {
+			if errors.Is(err, fs.ErrNotExist) {
+				// Case the file does not exist yet, we loop until it exist.
+				notFoundMsg := sessionContextMessage + "Cannot open in follow mode the container logs " + containerOutputPath + " because it does not exist yet, sleeping before retrying..."
+				log.G(h.Ctx).Debug(notFoundMsg)
+				// Warning: if we don't write anything to body before 30s, there will be a timeout in VK to API DoReq(), thus we send an informational message that the log is not ready.
+				w.Write([]byte(notFoundMsg))
+				// Flush otherwise it will be as if nothing was written.
+				if f, ok := w.(http.Flusher); ok {
+					log.G(h.Ctx).Debug(sessionContextMessage, "wrote file not found yet, now flushing this message...")
+					f.Flush()
+				} else {
+					log.G(h.Ctx).Error(sessionContextMessage, "wrote file not found but could not flush because server does not support Flusher.")
+				}
+				time.Sleep(4 * time.Second)
+				continue
+			} else {
+				// Case unknown error.
+				errWithContext := fmt.Errorf(sessionContextMessage+"could not open file to follow logs at %s error type: %s error: %w", containerOutputPath, fmt.Sprintf("%#v", err), err)
+				log.G(h.Ctx).Error(errWithContext)
+				w.Write([]byte(errWithContext.Error()))
+				return err
+			}
+		}
+		// File exist.
+		log.G(h.Ctx).Debug(sessionContextMessage, "opened for follow mode the container logs ", containerOutputPath)
+		break
+	}
+	defer containerOutputFd.Close()
+
+	// We follow only from after what is already read.
+	_, err = containerOutputFd.Seek(int64(containerOutputLastOffset), 0)
+	if err != nil {
+		errWithContext := fmt.Errorf(sessionContextMessage+"error during Seek() of GetLogsFollowMode() in GetLogsHandler of file %s offset %d type: %s %w", containerOutputPath, containerOutputLastOffset, fmt.Sprintf("%#v", err), err)
+		w.Write([]byte(errWithContext.Error()))
+		return errWithContext
+	}
+
+	containerOutputReader := bufio.NewReader(containerOutputFd)
+
+	bufferBytes := make([]byte, 4096)
+
+	// Looping until we get end of job.
+	// TODO: handle the Ctrl+C of kubectl logs.
```

Comment:
how long the session last anyway? I think if left idle it will be cut after 120 sec now. So it's probably something that would be worriesome only if we will ever manage to get 10k logs -f requests at the same time, and all of them are interupted in a 120s range.. quite rare, can I ask you to open an issue for this? TODOs are easily forgotten

Reply:
A session could last indefinitely, as long as the container is running.

Could you tell me where is coming this 120s? I will retry but I think in the current implementation, the kubectl logs -f can run indefinitely, as long as the container is still running.

The size of logs is not really problematic because they are being streamed, so no big memory is needed in interlink slurm plugin.
What could be problematic is if someone starts to do plenty of kubectl logs, so it will stress the filesystem a bit. But there is not much we can do except changing on the sleep period between reads.

Reply:
about the todo, are you referring to the one related to when someone does ctrl+c of during a `kubectl logs -f`?

Reply:
Yes, the todo I mean the ctl+c. 120 sec will be the interLink client configuration, so from its side the connection will close. The "tail" on the plugin will persist, I see, but it's worth a separate issue to understamd if we can track the closed connection and send a kill to the tail session.

Reply:
I tested just now. I have a pod that logs errors continuously. Even after 3 minutes, I could see the logs being followed.
So I don't understand what or where is this 120s hard-coded somewhere?

Reply:
but yes for a separate issue to have ctrl+c being handled to save resource.

Reply:
> I tested just now. I have a pod that logs errors continuously. Even after 3 minutes, I could see the logs being followed.
> So I don't understand what or where is this 120s hard-coded somewhere?

It's not in release yet. Any how, it should not count as Idle connection while waiting for logs. It should only impact the ctrl+c like scenario. I am giving it a try tomorrow, so far I confirm thst all other tests I did looks good! Awesome job!

Reply:
New issue about TODO: https://github.com/interTwin-eu/interlink-slurm-plugin/issues/40

Also for info, I tested 15min of kubectl logs -f, it is working without issue.